web/src/components/pages/dashboard/stats/leaderboard-card.tsx

```diff
@@ -44,12 +45,16 @@ const FetchRow = (props: React.HTMLProps<HTMLButtonElement>) => (
 interface State {
   rows: { position: number; [key: string]: any }[];
   isAtEnd: boolean;
+  avatarClipPlaying: boolean;
+  clipPosition: number;
```

Comment:
I think the state could be reduced to a single property, `playingClipIndex?: number`. And if it's null, no avatar clip is playing. Or am I missing something?